rules.d/60-block.rules

```diff
@@ -9,5 +9,5 @@ ACTION=="change", SUBSYSTEM=="scsi", ENV{DEVTYPE}=="scsi_device", TEST=="block",
 
 # watch metadata changes, caused by tools closing the device node which was opened for writing
 ACTION!="remove", SUBSYSTEM=="block", \
-  KERNEL=="loop*|mmcblk*[0-9]|msblk*[0-9]|mspblk*[0-9]|nvme*|sd*|vd*|xvd*|bcache*|cciss*|dasd*|ubd*|ubi*|scm*|pmem*|nbd*|zd*", \
+  KERNEL=="loop*|mmcblk*[0-9]|msblk*[0-9]|mspblk*[0-9]|nvme*|sd*|vd*|xvd*|bcache*|cciss*|dasd*|ubd*|ubi*|scm*|pmem*|nbd*|zd*|rbd*", \
   OPTIONS+="watch"
```

Comment:
I am not familiar, but is it safe to watch close inotify of the device node?

Reply:
The CEPH own rules only provide additional symlinks to their device. But the ones in systemd's persistent storage rules revolve around the blkid result mostly, which is about the layering above (a filesystem that resides on the block device, any other detected virtual block device layer...).

Reply:
> is it safe to watch close inotify of the device node?

I'm not sure what you mean by "safe", but we have definitely had to deal with this watch mechanism interfering with `rbd device unmap` command which tears down the device in the past.  Our tests would open the device, write something to it, close it and immediately proceed to unmapping it only for `rbd device unmap` to fail with EBUSY error because the RBD kernel driver checks for openers (by default -- this check can be overridden).  Our users reported hitting this scenario too.  This was back in 2013 so I'm positive that it was enabled for RBD: 

https://github.com/ceph/ceph/commit/8ca2274cc09c6a9925cbe86ccaf68db7898956e8

It appears that this mechanism was disabled for RBD shortly after systemd 219, probably with this commit:

https://github.com/systemd/systemd/commit/1aff20687f486857574fde0e5946a80b8ec212ba

AFAIU what this PR is doing is just resurrecting it, after ~9 years.  Do you see anything else going on here?

Reply:
> The CEPH own rules only provide additional symlinks to their device.

Right, this symlink would contain a Ceph pool name (possibly including a namespace inside of that pool) and an RBD image or snapshot name.  It's orthogonal to what blkid-based rules would do.